src/js/trove/multiple-regression.js

```diff
@@ -0,0 +1,182 @@
+  provides: {
+    values: {
+      "multiple-regression": "tany"
+    }
+  },
+  theModule: function(runtime, namespace, uri) {
+
+    function makeColumnMatrix(y_s) {
+      let n = y_s.length;
+      let M = new Array(n);
+      for (let i = 0; i < n; i++) {
+        M[i][0] = y_s[i];
+      }
+      return M;
+    }
+
+    function matrixTranspose(m) {
+      let numrows = m.length;
+      let numcols = m[0].length;
+      let mT = new Array(numcols);
+      for (let i = 0; i < numcols; i++) {
+        mT[i] = new Array(m);
+        for (let j = 0; j < numrows; j++) {
+          mT[i][j] = m[j][i];
+        }
+      }
+      return mT;
+    }
+
+    function matrixDeterminant(m) {
+      let size = m.length;
+      if (size === 0) {
+        return 0;
+      }
+      if (m[0].length !== size) {
+        return 0;
+      }
+      if (size === 1) {
+        return m[0][0];
+      }
+      if (size === 2) {
+        return (m[0][0] * m[1][1] - m[1][0] * m[0][1]);
+      }
+      let det = 0;
+      for (let i = 0; i < size; i++) {
+        det += m[0][i] * matrixCofactor(m, 0, i);
+      }
+      return det;
+    }
+
+    function matrixCofactor(m, i, j) {
+      let size = m.length;
+      if (size === 0) {
+        return 0;
+      }
+      if (m[0].length !== size) {
+        return 0;
+      }
+      let mm = new Array(size - 1);
+      for (let ii = 0; ii < size; ii++) {
+        if (ii === i) { continue; }
+        let iii = (ii > i ? ii - 1 : ii)
+        mm[iii] = new Array(size - 1);
+        for (let jj = 0; jj < size; jj++) {
+          if (jj === j) { continue; }
+          let jjj = (jj > j ? jj - 1 : jj)
+          mm[iii][jjj] = m[ii][jj];
+        }
+      }
+      let sign = (i + j) % 2;
+      sign = (sign === 0) ? 1 : -1;
+      return sign * matrixDeterminant(mm)
+    }
+
+    function matrixAdjoint(m) {
+      let size = m.length;
+      let mc = new Array(size);
+      for (let i = 0; i < size; i++) {
+        mc[i] = new Array(size);
+        for (let j = 0; j < size; j++) {
+          mc[i][j] = matrixCofactor(m, i, j);
+        }
+      }
+      return matrixTranspose(mc)
+    }
+
+    function matrixInverse(m) {
+      let det = matrixDeterminant(m);
+      let mI = matrixAdjoint(m);
+      let k = m.length;
+      for (let i = 0; i < k; i++) {
+        for (let j = 0; j < k; j++) {
+          let x = mI[i][j]
+          mI[i][j] = x/det
+        }
+      }
+      return mI;
+    }
+
+    function matrixMultiply2(m1, m2) {
+      let m1r = m1.length;
+      let m1c = m1[0].length;
+      let m2c = m2[0].length;
+      let mP = new Array(m1r);
+      for (let i = 0; i < m1r; i++) {
+        mP[i] = new Array(m2c);
+        for (let j = 0; j < m2c; j++) {
+          let x = 0;
+          for (let k = 0; k < m1c; k++) {
+            x += m1[i][k] * m2[k][j];
+          }
+          mP[i][j] = x;
+        }
+      }
+      return mP;
+    }
+
+    function multipleRegression(x_s_s, y_s) {
```

Comment:
For the non-statisticians among us, it would be helpful to include some comments explaining the formulas a bit and/or links to resources (e.g. https://en.wikipedia.org/wiki/Linear_regression#Least-squares_estimation_and_related_techniques) that define the formulas, otherwise seeing things like random `Xi[0] = 1` floating around just feels inscrutable.